src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -817,7 +831,12 @@ private void initChannelPipeline(ChannelPipeline pipeline) {
                 new IdleStateHandler(0, 0, proxyServer
                         .getIdleConnectionTimeout()));
 
+        if (proxyServer.getGlobalStateHandler() != null) {
+            pipeline.addLast("outboundGlobalStateHandler", new OutboundGlobalStateHandler(this));
+        }
+
         pipeline.addLast("handler", this);
+
     }
```

Comment:
`OutboundGlobalStateHandler` is placed after the `handler` as the `handler` will not take part in outbound message processing as it is of inbound type. It just writes data to the channel which starts the outbound processing where `OutboundGlobalStateHandler ` is the first in the chain.